### PR TITLE
feat: Install PostHog and implement data capture on various buttons

### LINKED ISSUE #1430: Implement PostHog
Screen Shot 2022-05-05 at 4 26 28 PM

https://app.posthog.com/ingestion/web

## Steps
1. Implement the node posthog
2. Add event for login only
3. Open PR

### LINKED ISSUE #1434: Implement Post Hog into onboarding buttons

### PR DESCRIPTION
**Pull request by chadstewart**
## What type of PR is this? (check all applicable)

- [x] 🍕 Feature
- [ ] 🐛 Bug Fix
- [ ] 📝 Documentation Update
- [ ] 🎨 Style
- [ ] 🧑‍💻 Code Refactor
- [ ] 🔥 Performance Improvements
- [ ] ✅ Test
- [ ] 🤖 Build
- [ ] 🔁 CI
- [ ] 📦 Chore (Release)
- [ ] ⏩ Revert

## Description

This PR [adds] the [feature] posthog logging. 

## Related Tickets & Documents

Resolves #1430 and resolves #1434

## Added tests?

- [ ] 👍 yes
- [x] 🙅 no, because they aren't needed
- [ ] 🙋 no, because I need help

## Added to documentation?

- [ ] 📜 README.md
- [ ] 📓 docs.opensauced.pizza
- [ ] 🍕 dev.to/opensauced
- [ ] 📕 storybook
- [x] 🙅 no documentation needed

### CODE REVIEW

File: src/index.jsx [hunk 1]
```diff
@@ -24,6 +25,8 @@ function Index() {
   const [isAdmin, setIsAdmin] = useState(null);
 
   useEffect(() => {
+    posthog.init('phc_QFGGi71ugzUOXTLUTwMHVTlaHhDBolEC3XNVHNFHKmz', { api_host: 'https://app.posthog.com' });
```
Comment (bdougie) [inline src/index.jsx:28]: Might need to put this into ENV var

**Comment by 0-vortex** [comment]
For #1430 a very small wrapper could be useful, currently, we init PostHog in `index.jsx`, and then `.capture` works everywhere else. To make it easier to catch errors and standardize the way we send events from components, moving that code into something like `lib/posthog` or `lib/analytics` would centralize the way we implement PostHog across the app. 

Other than that and extracting the environment variable to `src/env.d.ts` / `.env` the code LGTM! Would still leave #1430 and do it separately from this PR

**Comment by chadstewart** [comment]
> For #1430 a very small wrapper could be useful, currently, we init PostHog in `index.jsx`, and then `.capture` works everywhere else. To make it easier to catch errors and standardize the way we send events from components, moving that code into something like `lib/posthog` or `lib/analytics` would centralize the way we implement PostHog across the app.
> 
> Other than that and extracting the environment variable to `src/env.d.ts` / `.env` the code LGTM! Would still leave #1430 and do it separately from this PR

That's a good idea, making the wrapper function. I'll go work on that now.

So the reason why I added both the issues to this PR was because the implement Posthog PR also required implementing data capture on the login button so I figured since I was adding it there, I might as well just add it to the onboarding buttons here rather than make a whole other PR. Figured referencing both issues and also having separate commits would give enough context.

**Review by bdougie — APPROVED** [review]
LGTM

**Comment by bdougie** [comment]
> 
> Making the `phc_` token an environment variable's only purpose is to be able to rotate it across environments, it would still be technically public so the `VITE_` notation should be fine, and committing to both env files if you like. This will make a little bit more sense when we apply the code you write in this PR into hot.opensauced 🍕 

I would love to also capture this in hot.opensauced

**Pull request merged**